cmake-format.yaml

```diff
@@ -0,0 +1,246 @@
+_help_parse: Options affecting listfile parsing
+parse:
+  _help_additional_commands:
+    - Specify structure for custom cmake functions
+  additional_commands:
+    foo:
```

Comment:
[fix] Presumably some leftover copy-paste example?

Reply:
No, this cmake-format.yaml is a dump `cmake-lint --dump-config`. I think we should leave it as it stands rather than cutting it down, considering the default structure is autogenerated.

Reply:
I think I side with David - having this here blurs what is intentional vs illustrative boilerplate. If it was commented out, that might be different. I totally get having parameters not left to implicit defaults, but active foo-barry seems to offer no real benefit. Would it make more sense to use this section to actually define our custom commands? (Assuming I understand its purpose correctly)

Reply:
Removed the additional command example.

https://github.com/OpenAssetIO/OpenAssetIO/pull/811/commits/b6ca58463e247f22e3b5ef0e59f19f6d2cf39f0e

Reading the [docs](https://cmake-format.readthedocs.io/en/latest/format-features.html#custom-commands), it seems that the intent here is to handle custom commands defined by `add_custom_command` ,which we don't have.

This section may or may not be intended to also manage `add_custom_target`, which we have a lot of, the docs don't make that clear. I didn't manage to produce any formatting variations adding them there. Given this, I'd rather not get really drawn in trying to figure out this niche concern, and leave this section blank.

Reply:
Yeah the docs aren't super clear.  Based on the example, I'm pretty certain this is a way to lint `function`s that use [`cmake_parse_arguments`](https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html) (we don't currently use that, but we really should for some of our functions that take multiple arguments).